src/api/helpers/get-populated-application/index.ts

```diff
@@ -99,11 +119,26 @@ const getPopulatedApplication = async ({
       totalContractValueOverThreshold,
     };
 
+    /**
+     * Check that all expected relationships are populated.
+     * If not, throw an error.
+     * Otherwise, unexpected behaviours could occur.
+     */
+    Object.keys(populatedApplication).forEach((relationshipKey: string) => {
+      if (EXPECTED_RELATIONSHIPS.includes(relationshipKey)) {
+        const populatedRelationship = populatedApplication[relationshipKey];
+
+        if (!populatedRelationship || !objectHasKeysAndValues(populatedRelationship)) {
```

Comment:
I would expect this function to evaluate whether `objectHasKeysAndValues` is falsy or not.

Reply:
This function only needs to check for false/negative scenarios i.e:

- `populatedApplication[relationship]` is false
- `populatedApplication[relationship]` is an object without keys or values 

We do not need to check if `objectHasKeysAndValues` is true, the function is purely for throwing an error (for debugging), if either of the above scenarios are met

Reply:
I think the function can incorporate falsy argument check before validating above mentioned scenarios and then you if condition will be a simple function call.

Reply:
I'm not following what you mean @abhi-markan can you provide an example please? The function is checking if any relationship is false or has no key/values.

I do not see how the implementation could be cleaner/simpler

Reply:
It's checking the nested properties in `populatedApplication` i.e

```ts
const populatedApplication = {
  relationshipA, 
  relationshipB,
  relationshipC,
  // etc
};
```

So this line:

```ts
if (!populatedRelationship || !objectHasKeysAndValues(populatedRelationship)) {
```

Is checking relationshipA, relationshipB etc

Reply:
Updated as discussed 👍